File: mobile/test/core/model/zone_resolver_test.dart
import 'package:meshsetu_mobile/core/model/model.dart';
import 'package:test/test.dart';

void main() {
  test('nearest fresh anchor wins and missing anchor degrades', () {
    final resolver = ZoneResolver({
      'a': const ZoneAnchor(anchorId: 'a', logicalZone: 'Gate A'),
      'b': const ZoneAnchor(anchorId: 'b', logicalZone: 'Gate B'),
    });
    expect(
      resolver.estimate([
        const BeaconObservation(anchorId: 'a', rssi: -80, observedAtMs: 100),
        const BeaconObservation(anchorId: 'b', rssi: -50, observedAtMs: 100),
      ], 100).logicalZone,
      'Gate B',
    );
    expect(resolver.estimate(const [], 100).uncertainty, 'unknown');
  });
}

File: core-model/src/test/kotlin/in/meshsetu/model/ZoneResolverTest.kt
package `in`.meshsetu.model

import kotlin.test.Test
import kotlin.test.assertEquals

class ZoneResolverTest {
    @Test fun nearestFreshAnchorWinsAndMissingAnchorDegrades() {
        val resolver = ZoneResolver(mapOf("a" to ZoneAnchor("a", "Gate A"), "b" to ZoneAnchor("b", "Gate B")))
        assertEquals("Gate B", resolver.estimate(listOf(BeaconObservation("a", -80, 100), BeaconObservation("b", -50, 100)), 100).logicalZone)
        assertEquals("unknown", resolver.estimate(emptyList(), 100).uncertainty)
    }
}

